### PR TITLE
[O11y][Apache Spark] Resolve the conflicts in host.ip field

### LINKED ISSUE #7407: [O11y][Apache Spark] Conflicts in host.ip field
In Apache Spark, found that host.ip field has conflicts because `host.ip` is not mentioned in the `ecs.yml`. Please refer to the below screenshot.

conflict

- [x] Need to add the `host.ip` field to the ecs.yml file.
- [x] Add data re-indexing steps to the README.

## Related issue

- https://github.com/elastic/integrations/issues/7406

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Bug

## What does this PR do?

- Added host.ip field in ecs.yml and added re-indexing steps to resolve conflicts for existing users.
- ECS fields are being used in the pipeline of the particular data streams, but in the ecs.yml file the used fields are not mentioned. Hence need to update the ecs.yml file with additional fields and sort the ecs.yml in alphabetical order.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## Related issues

- Closes #7407
- Relates #7406

### CODE REVIEW

**Review by milan-elastic — APPROVED** [review]
LGTM

**Comment by muthu-mps** [comment]
@harnish-elastic  - Update the description on the reason for adding additional ECS fields and the context for renaming a few ECS fields.

**Comment by harnish-elastic** [comment]
> * ECS fields

@muthu-mps Updated the description [here](https://github.com/elastic/integrations/pull/7468#:~:text=for%20existing%20users.-,ECS%20fields,-are%20being%20used). Also regarding renaming of ecs fields, I have just sorted the ecs.yml file in the alphabetical order and not renamed any ecs fields!

File: packages/apache_spark/_dev/build/docs/README.md [hunk 1]
```diff
@@ -63,6 +63,100 @@ Restart Spark master.
 
 Follow the same set of steps for Spark Worker, Driver and Executor.
 
+### Troubleshooting
+
+If host.ip is shown conflicted under ``metrics-*`` data view, then this issue can be solved by reindexing the ``Application``, ``Driver``, ``Executor`` and ``Node`` data stream's indices.
```
Comment (ishleenk17) [inline packages/apache_spark/_dev/build/docs/README.md:68]: The reindexing step is important for this change, but I am discussing it within team if we need to give this long troubleshooting steps in the README. 

cc: @lalit-satapathy

Comment (lalit-satapathy) [reply packages/apache_spark/_dev/build/docs/README.md:68]: >The reindexing step is important for this change, but I am discussing it within team if we need to give this long troubleshooting steps in the README.

Long term, we can explore remove such repetitive sections in the package Readme and capture in a different way?

File: packages/apache_spark/data_stream/application/fields/ecs.yml [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 - external: ecs
   name: event.kind
 - external: ecs
-  name: event.type
+  name: event.module
 - external: ecs
-  name: ecs.version
+  name: event.type
 - external: ecs
-  name: tags
```
Comment (ishleenk17) [inline packages/apache_spark/data_stream/application/fields/ecs.yml:8]: any advantage of changing the ordering of this mapping ?

Comment (harnish-elastic) [reply packages/apache_spark/data_stream/application/fields/ecs.yml:8]: This is one of the best practices to have field names in the alphabetical order for readability.

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Review by muthu-mps — APPROVED** [review]
Looks good!

**Pull request merged**